hibernate-core/src/main/java/org/hibernate/generator/OnExecutionGenerator.java

```diff
@@ -114,18 +116,23 @@ public interface OnExecutionGenerator extends Generator {
 	 * {@link org.hibernate.id.IdentityGenerator}. And the need for customized behavior for
 	 * identity columns is the reason why this layer-breaking method exists.
 	 */
-	@Incubating
```

Comment:
I guess promoting this to non-incubating might be too early, no?

Reply:
Absolutely, I had substituted `@Incubating` with `@Deprecated` but then removed the annotation without restoring the old one. Thanks for checking.